packages/realm-web/src/App.test.ts

```diff
@@ -282,4 +292,163 @@ describe("App", () => {
         expect(app.services).keys(["mongodb", "http"]);
         expect(typeof app.services.mongodb).equals("function");
     });
+
+    it("hydrates users from storage", () => {
+        const storage = new MemoryStorage();
+        const transport = new MockNetworkTransport([]);
+
+        // Fill data into the storage that can be hydrated
+        const appStorage = storage.prefix("app(default-app-id)");
+        appStorage.set("userIds", JSON.stringify(["alices-id", "bobs-id"]));
+
+        const alicesStorage = appStorage.prefix("user(alices-id)");
+        alicesStorage.set("accessToken", "alices-access-token");
+        alicesStorage.set("refreshToken", "alices-refresh-token");
+        alicesStorage.set(
+            "profile",
+            JSON.stringify({
+                type: "normal",
+                identities: [],
+                firstName: "Alice",
+            }),
+        );
+
+        const bobsStorage = appStorage.prefix("user(bobs-id)");
+        bobsStorage.set("accessToken", "bobs-access-token");
+        bobsStorage.set("refreshToken", "bobs-refresh-token");
+
+        const app = new App({
+            id: "default-app-id",
+            storage,
+            transport,
+            baseUrl: "http://localhost:1337",
+        });
+
+        expect(app.allUsers.length).equals(2);
+
+        const alice = app.allUsers[0];
+        expect(alice.id).equals("alices-id");
+        expect(alice.accessToken).equals("alices-access-token");
+        expect(alice.refreshToken).equals("alices-refresh-token");
+        expect(alice.profile.firstName).equals("Alice");
+
+        const bob = app.allUsers[1];
+        expect(bob.id).equals("bobs-id");
+        expect(bob.accessToken).equals("bobs-access-token");
+        expect(bob.refreshToken).equals("bobs-refresh-token");
+    });
```

Comment:
Actually - its not optional, but it would throw if requested, because we explicitly ask for it not to fetch the profile when we login (second argument of the `logIn` method).

Reply:
Fair enough, do we test that it throws?

Reply:
Yep 🤞  https://github.com/realm/realm-js/blob/kh/web/persist-users/packages/realm-web/src/User.test.ts#L82-L84